lib/cc/analyzer/engines_config_builder.rb

```diff
@@ -11,11 +11,11 @@ class EnginesConfigBuilder
         :container_label,
       )
 
-      def initialize(registry:, config:, container_label:, source_dir:, include_paths:)
+      def initialize(registry:, config:, container_label:, source_dir:, include_paths_builder:)
```

Comment:
Ah, gotcha, missed the singleton part.

Would it be hard to do this in builder? I think all you need to do is pass in:

```ruby
IncludePathsBuilder.new(
  @config.exclude_paths || [], @requested_paths
)
```

If not this might be worth moving to a singleton if that works for builder.